### PR TITLE
Handle previous increment in all cases in readless increment co-processor

### PR DESCRIPTION
**Pull request by poornachandra**
JIRA contains details of the fix - https://issues.cask.co/browse/CDAP-3449

### CODE REVIEW

**Comment by chtyim** [comment]
Do you have more details about what the bug is about? Also, does it only affect 0.98?

**Comment by poornachandra** [comment]
Sorry, missed including the JIRA in the description. This fix has to be ported to all other versions too. This is PR is for initial review.

**Comment by poornachandra** [comment]
Updated the PR with the changes to all versions of HBase we support.

**Comment by poornachandra** [comment]
Build - http://builds.cask.co/browse/CDAP-DUT2560-4

File: cdap-hbase-compat-0.96/src/main/java/co/cask/cdap/data2/increment/hbase96/IncrementSummingScanner.java [hunk 1]
```diff
@@ -221,14 +234,18 @@ private boolean nextInternal(List<Cell> cells, int limit) throws IOException {
     // emit any left over increment, if we hit the end
     if (previousIncrement != null) {
       // in any situation where we exited due to limit, previousIncrement should already be null
-      Preconditions.checkState(limit <= 0 || addedCnt < limit);
+      Preconditions.checkState(limit <= 0 || addedCnt < limit, "addedCnt={}, limit={}", addedCnt, limit);
```
Comment (bdmogal) [inline cdap-hbase-compat-0.96/src/main/java/co/cask/cdap/data2/increment/hbase96/IncrementSummingScanner.java]: The format specifier for Preconditions.checkState is ``%s`` - http://guava-libraries.googlecode.com/svn-history/r21/trunk/javadoc/com/google/common/base/Preconditions.html#checkState(boolean, java.lang.String, java.lang.Object...)

Comment (poornachandra) [inline cdap-hbase-compat-0.96/src/main/java/co/cask/cdap/data2/increment/hbase96/IncrementSummingScanner.java]: Thanks for catching this!

**Comment by poornachandra** [comment]
Build - http://builds.cask.co/browse/CDAP-DUT2560-5

**Comment by henry-cask** [comment]
LGTM

**Comment by poornachandra** [comment]
Build passed, commits squashed.

**Pull request merged**